### PR TITLE
Service: Matrix Keypad

### PR DESCRIPTION
**Pull request by pelikhan**
This pull request adds a new service definition for JACDAC.

### CODE REVIEW

File: services/matrixkeypad.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+## Registers
+
+    ro pressed @ reading {
+        repeats:
+          index: u8
+    }
+
+The coordinate of the button currently pressed. Keys are zero-indexed from left to right, top to bottom:
+``row = index / columns%, ``column = index % columns``.
```
Comment (mmoskal) [inline services/matrixkeypad.md:16]:
```suggestion
``row = index / columns``, ``column = index % columns``.
```

File: services/matrixkeypad.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    const rows: u8 @ 0x180
+
+Number of rows in the matrix
+
+    const columns: u8 @ 0x181
+
+Number of columns in the matrix
+
+    const keys?: string @ 0x182
```
Comment (pelikhan) [inline services/matrixkeypad.md:26]:
```
{
repeats:
    string
}
```

File: services/matrixkeypad.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+Number of rows in the matrix
+
+    const columns: u8 @ 0x181
+
+Number of columns in the matrix
+
+    const labels? @ 0x182 {
+        repeats:
+            label: string
```
Comment (mmoskal) [inline services/matrixkeypad.md:28]: this should be `string0`

Comment (mmoskal) [reply services/matrixkeypad.md:28]: fixing

**Comment by pelikhan** [comment]
Can the compiler catch those issues?

________________________________

File: services/matrixkeypad.md (hunk 3 above)
Comment (mmoskal) [reply services/matrixkeypad.md:28]: > Can the compiler catch those issues?
on my todo list

**Pull request merged**